RecoParticleFlow/PFClusterProducerAlpaka/plugins/alpaka/PFClusterProducerAlpaka.cc

```diff
@@ -0,0 +1,80 @@
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitHostCollection.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/alpaka/PFClusterParamsAlpakaESData.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/PFClusterParamsAlpakaESRecord.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/alpaka/PFClusterProducerKernel.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+
+#define DEBUG false
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class PFClusterProducerAlpaka : public stream::EDProducer<> {
+  public:
+    PFClusterProducerAlpaka(edm::ParameterSet const& config)
+        : pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
+          InputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("PFRecHitsLabelIn"))},
+          OutputPFClusterSoA_Token_{produces()},
+          OutputPFRHFractionSoA_Token_{produces()},
+          synchronise(config.getParameter<bool>("synchronise")),
+          _produceSoA{config.getParameter<bool>("produceSoA")},
+          _produceLegacy{config.getParameter<bool>("produceLegacy")} {}
+
+    void produce(device::Event& event, device::EventSetup const& setup) override {
+      const reco::PFClusterParamsAlpakaESDataDevice& params = setup.getData(pfClusParamsToken);
+      const reco::PFRecHitHostCollection& pfRecHits = event.get(InputPFRecHitSoA_Token_);
+      const int nRH = pfRecHits->size();
+
+      reco::tmpDeviceCollection tmp{nRH + 1, event.queue()};
+      reco::tmpEdgeDeviceCollection tmpEdge{(nRH * 8) + 1, event.queue()};
+      reco::PFClusterDeviceCollection pfClusters{nRH, event.queue()};
+      reco::PFRecHitFractionDeviceCollection pfrhFractions{nRH * 120, event.queue()};
+
+      if (!kernel)
+        kernel.emplace(PFClusterProducerKernel::Construct(event.queue(), pfRecHits));
```

Comment:
I see the `Construct()` function allocates temporary(?) Alpaka buffers that are held as member data. It would be better to allocate temporary buffers separately for each event to allow sharing them across EDModules.